### PR TITLE
Config changes

### PR DESCRIPTION
**Pull request by diff-by-default**
A few more changes mostly to make this easier for me to use. 

You can now specify a file extension with `-e png` etc. You can also now run it without the gui showing everything that is going on with `--headless`

### CODE REVIEW

**Comment by diff-by-default** [comment]
Also, sorry for the unneeded commits: a76ab2a835458b4d7735af62b4f83f0cd4d512c1 0ff85137271e53449b99015a0291f17f7d143a17

Apparently I'm still not as good at github as I though I was... It keeps wanting to make me PR the commits that were me attempting to sync with your repo...or something. Who knows.

File: pokemap.py [hunk 1]
```diff
@@ -18,15 +18,26 @@ def main():
...
+    os.environ["SDL_VIDEODRIVER"] = "dummy" #this works on my ubuntu machine, but untested on others.
+
   global DEBUG_MODE
   if args.verbose:
     DEBUG_MODE = True
 
   bytes = load_rom(args.rom_file)
   strings = load_strings(bytes, '0x3eecfc')
-  debug('Found all these strings: {}'.format(strings))
+  #debug('Found all these strings: {}'.format(strings))
```
Comment (mmmulani) [inline pokemap.py]: can you uncomment this?

File: pokemap.py [hunk 2]
```diff
@@ -44,12 +55,14 @@ def main():
   screen = pygame.display.set_mode((width, height))
   screen.fill((255, 255, 255))
 
+  if args.headless: print("Working!...")
```
Comment (mmmulani) [inline pokemap.py]: minor nit: can you put all if's on two lines, e.g.

```
if args.headless:
  print(..)
```

**Comment by diff-by-default** [comment]
Completely forgot I commented out that line! Should be fixed now. I was getting some encoding issues on my machine originally, so I had lazily just killed the line instead of actually fixing it >.>

File: pokemap.py [hunk 3]
```diff
@@ -5,23 +5,25 @@
...
   parser.add_argument('rom_file', metavar='rom', type=str,
                        help='a fire red rom')
-  parser.add_argument("-e","--extension",
-                      help="Specify the output file extension",
-                      choices=['png','jpeg','tga','bmp'],
-                      default='bmp')
+  parser.add_argument("-o","--outfile",
+                      help="Specify the output file name/extension",
+                      action=CheckExt({'png','bmp','jpeg','tga'}),
+                      default='wholemap.bmp')
```
Comment (diff-by-default) [inline pokemap.py:26]: This serves the same purpose, but is just more robust and allows for changing name+ext

**Comment by mmmulani** [comment]
thanks for the work! just curious, what are you using this for now?

**Comment by diff-by-default** [comment]
Just messing around with some stuff right now, it hasn't actually taken any sort of shape yet. I did throw a little map together using your script and gdal2tiles though, you can see a little demo [here](http://dev.jargonaut.net/kanto/map.html)

Now I've just got to figure out how the ds roms work to get tiles for the other regions...

**Comment by mmmulani** [comment]
ah that looks awesome!

some things I was considering:
- adding support for the other game
- rendering NPCs on the map for fun
- maybe trying to render the animations

**Comment by diff-by-default** [comment]
I added some relevant issues for those (those were actually the two things I was thinking about next also). The animation rendering is...interesting. I'd be very interested into looking into constructing a gif or something similar and attempting to render it in openlayers or another mapping tool.

**Pull request merged**